src/scheduler.rs

```diff
@@ -5,25 +5,29 @@ use std::collections::VecDeque;
 use std::rc::Rc;
 use std::sync::atomic::{AtomicBool, Ordering};
 
-pub(crate) type Shared<T> = Rc<RefCell<T>>;
+/// Provides a mutable reference counted value
+pub type Shared<T> = Rc<RefCell<T>>;
```

Comment:
These visibility changes were (AFAIK) necessary to allow the tests to import the scheduler type.

Reply:
You might want to slap a [#[doc(hidden)]](https://doc.rust-lang.org/rustdoc/the-doc-attribute.html#dochidden) on that then to prevent it from being included in documentation.